internal/controllers/watcher/controller.go

```diff
@@ -207,3 +230,50 @@ func (r *WatcherReconciler) manifestToObject(m workv1.Manifest) (client.Object,
 
 	return clientObj, nil
 }
+
+var hostedClusterPredicate = builder.WithPredicates(predicate.Funcs{
+	UpdateFunc: func(e event.UpdateEvent) bool {
+		newHC := e.ObjectNew.(*hyperv1.HostedCluster)
+		oldHC := e.ObjectOld.(*hyperv1.HostedCluster)
+		return newHC.Spec.ClusterID == oldHC.Spec.ClusterID
+	},
+	CreateFunc:  func(e event.CreateEvent) bool { return true },
+	DeleteFunc:  func(e event.DeleteEvent) bool { return true },
+	GenericFunc: func(e event.GenericEvent) bool { return false },
+})
+
+var hypershiftServiceMonitorsPredicate = builder.WithPredicates(predicate.Funcs{
+	UpdateFunc:  func(e event.UpdateEvent) bool { return isHypershiftServiceMonitor(e.ObjectNew) },
+	CreateFunc:  func(e event.CreateEvent) bool { return isHypershiftServiceMonitor(e.Object) },
+	DeleteFunc:  func(e event.DeleteEvent) bool { return isHypershiftServiceMonitor(e.Object) },
+	GenericFunc: func(e event.GenericEvent) bool { return false },
+})
+
+func isHypershiftServiceMonitor(obj client.Object) bool {
+	if obj.GetName() == mconfig.HypershiftEtcdServiceMonitorName || obj.GetName() == mconfig.HypershiftApiServerServiceMonitorName {
+		return isHypershiftOwned(obj)
+	}
+
+	if obj.GetName() == mconfig.AcmEtcdServiceMonitorName || obj.GetName() == mconfig.AcmApiServerServiceMonitorName {
+		return true
+	}
+
+	return false
+}
+
+func isHypershiftOwned(obj client.Object) bool {
+	for _, owner := range obj.GetOwnerReferences() {
+		gv, err := schema.ParseGroupVersion(owner.APIVersion)
+		if err != nil {
+			continue
+		}
+
+		if gv.Group != hyperv1.GroupVersion.Group {
+			continue
+		}
+
+		return true
+	}
+
+	return false
+}
```

Comment:
Re wrote it slightly to be easier to follow. I would also pass a logger to this function to log when err != nil
```suggestion
	if obj.GetName() == mconfig.HypershiftEtcdServiceMonitorName || obj.GetName() == mconfig.HypershiftApiServerServiceMonitorName {
		for _, owner := range obj.GetOwnerReferences() {
			gv, err := schema.ParseGroupVersion(owner.APIVersion)
			if err != nil {
				continue
			}

			if gv.Group == hyperv1.GroupVersion.Group {
				return true
			}
		}
	}

	if obj.GetName() == mconfig.AcmEtcdServiceMonitorName || obj.GetName() == mconfig.AcmApiServerServiceMonitorName {
		return true
	}

	return false
}
```

Reply:
Good point for the logger, adding that.